Content.Shared/GameObjects/Components/Disposal/SharedDisposalRouterComponent.cs

```diff
@@ -0,0 +1,53 @@
+﻿#nullable enable
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.UserInterface;
+using Robust.Shared.Serialization;
+using System;
+
+namespace Content.Shared.GameObjects.Components.Disposal
+{
+    public class SharedDisposalRouterComponent : Component
+    {
+        public override string Name => "DisposalRouter";
+
+        [Serializable, NetSerializable]
+        public class DisposalRouterUserInterfaceState : BoundUserInterfaceState
+        {
+            public readonly string Tags;
+
+            public DisposalRouterUserInterfaceState(string tags)
+            {
+                Tags = tags;
+            }
+        }
+
+        [Serializable, NetSerializable]
+        public class UiActionMessage : BoundUserInterfaceMessage
+        {
+            public readonly UiAction Action;
+            public readonly string Tags = "";
```

Comment:
This should probably have a length limit imposed on it to prevent the user from inputting a giant regex string.